src/Agent.Listener/Agent.cs

```diff
@@ -76,6 +77,16 @@ public async Task<int> ExecuteCommand(CommandSettings command)
                     return Constants.Agent.ReturnCode.Success;
                 }
 
+                if (command.Diagnostics)
+                {
+                    PrintBanner();
+                    _term.WriteLine("Running Diagnostics Tests Only...");
+                    _term.WriteLine(string.Empty);
+                    DiagnosticTests diagnostics = new DiagnosticTests(_term);
+                    diagnostics.Execute();
+                    return Constants.Agent.ReturnCode.Success;
```

Comment:
I could argue both ways, but should we always return success or should the return code be dependent on whether all the checks pass? Would it make sense to have a command argument to control that?

Reply:
I thought about this too.  Here are the current return codes.

```
                public const int Success = 0;
                public const int TerminatedError = 1;
                public const int RetryableError = 2;
                public const int AgentUpdating = 3;
                public const int RunOnceAgentUpdating = 4;
```

I don't think we have one that fits for failures.  If someone wanted to put this in a pipeline though we should add a new code here.  What do you think?

Reply:
I am fine adding a new non-zero return code saying diagnostic test failed. Are you thinking option to to enable that feature?

This always returns 0
```
% run --diagnostics
```

This returns 0 if all checks pass, non-zero if any check fails
```
% run --diagnostics --exit-code-diag-result 
```
 ???

Reply:
> One other general question, do we need to localize the diagnostic message strings?
@jtpetty in past projects we haven't attempted to localize diagnostics information.  I think it's something we can add later if we feel necessary.

Reply:
Unless we see a specific need to have an 'error' exit code I'd rather leave it as is for now.

If there is a specific need to return an error on failure I think it probably makes sense to do that all the time and not have it be an option.

Reply:
Fair enough